### DISCUSSION TITLE
How to deal with typescript with property from pinia plugin in computed

### DISCUSSION THREAD

**Discussion by joris-gallot** [vuejs/pinia#2270] [Help and Questions] [answered]
Hey 👋 

Here is a basic store example with a plugin to understand my question
Plugin:
```ts
pinia.use(() => ({ taxe: ref(2) }));
```
Store:
```ts
const useMyStore = defineStore('my-store', () => {
  const self = useMyStore();
  const price = computed(() => 3 + self.taxe);

  return { price };
});
```
You can test it here: https://stackblitz.com/edit/vitejs-vite-sp4xr8?file=src%2FApp.vue

So my question is how to deal with this error because as you can see on stackblitz the price is calculated correctly but typescript isn't happy and now my store is typed as `any` 😔 ? has anyone ever had to deal with this?

![Capture d’écran 2023-06-20 à 15 13 41](https://github.com/vuejs/pinia/assets/70059224/1d0fbc31-7725-46a1-b178-1d92578332b5)

**Comment by posva** [comment, marked as answer]
This is not possible to type because types are recursive (the error you see)

I would suggest not to put the tax in a plugin but rather in a global variable or a store if needed

**Comment by joris-gallot** [comment]
well, in my case it's a bit more complicated, but thanks for the quick reply!
